File: 560. Subarray Sum Equals K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int size = nums.size(), count = 0;
        int PI[size];
        PI[0] = nums[0];
        for(int i = 1; i< size; i++){
            PI[i] = PI[i - 1] + nums[i];
        }
        for(int i = 0; i < size ; i++){
            if(PI[i] == k) count++;
                for(int j = i + 1; j < size; j++){
                    if(PI[j] - PI[i] == k){
                        count++;
                    }
                }
        }
        return count;

    }
};

File: 560. Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pref = []
        for num in nums:
            if len(pref) == 0:
                pref.append(num)
            else:
                pref.append(num + pref[-1])

        dic = defaultdict(int)
        c = 0

        for s in pref:
            if s == k:
                c += 1
            if s - k in dic:
                c += dic[s - k]
            dic[s] += 1
        return c

#1, 2, 3
#
       
            
